src/lib.js

```diff
@@ -142,6 +141,33 @@ mkDirExists = (ftp, dir) => {
     });
 };
 
+/**
+ * Applies the mapper function to each item in the array in series.
+ * @param {Array} array - The array to map over.
+ * @param {Function} mapper - The mapping function.
+ * @returns {Promise<Array>} A promise that resolves with the new array.
+ */
+function mapSeries(array, mapper) {
+    // Initialize a resolved Promise
+    let result = Promise.resolve();
+    // Empty array to store the results
+    const output = [];
```

Comment:
Could you implement this as a reduce please, then you won't need a global (within this fnction) variable

Reply:
I think this way is also ok, it is more clear than reduce

Reply:
I don't have anything against a `reduce` version. One thing makes me curious in this case: how it performs in cases of many items in the array. `reduce` is not the best in performance for loops, but I didn't want to go for the for-loop version. 😄

I tried to make it a little bit more readable than `reduce`, just like @a-tonchev mentioned.

Anyway, here is a suggestion code for the function:

```js
function mapSeries(array, mapper) {
    // Use the reduce method to chain promises together
    return array.reduce((promise, item, index) => {
        // For each item in the array, add a new promise to the chain
        return promise.then(output => {
            // Apply the mapper function to the current item
            return mapper(item, index, array).then(res => {
                // Push the result of the mapper function to the output array
                output.push(res);
                // Return the output array for the next iteration
                return output;
            });
        });
    // Start with a resolved promise and an empty array
    }, Promise.resolve([]));
}
```

It needs more comments to understand what exactly is done.

Reply:
@cyrdam one more thing - since bluebird is no more used, you can remove it from the package.json, so that we don't install not needed library there

Reply:
It is already removed: https://github.com/simonh1000/ftp-deploy/pull/167/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

Reply:
@simonh1000 do you want to have the [mentioned version](https://github.com/simonh1000/ftp-deploy/pull/167#discussion_r1430072436), or the current implementation from the PR?

Reply:
Ok, I've read about the reduce perf concerns. they probably don't apply at the scale of an ftp repo (and anyone with a very large directory would probably want the - as yet unimplemented - feature of upload only new files). But your map approach is not a deal breaker, so thank you for your efforts